pandas/tests/scalar/timedelta/test_construction.py

```diff
@@ -210,3 +210,27 @@ def test_td_constructor_on_nanoseconds(constructed_td, conversion):
 def test_td_constructor_value_error():
     with pytest.raises(TypeError):
         Timedelta(nanoseconds='abc')
+
+
+class not_raises(object):
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+
+@pytest.mark.parametrize("redundant_unit, expectation", [
+    ("", not_raises()),
```

Comment:
I believe `pytest.raises(None)` should suffice.

Reply:
It doesn't - it is still an open issue in pytest: https://github.com/pytest-dev/pytest/issues/1830

Reply:
Odd that we can have an empty context manager with warnings assertions, but that's not implemented for exceptions.  Implementing the empty context manager wouldn't super difficult on our end (we have `pandas.util.testing.assert_raises_regex`).

@jreback @jorisvandenbossche : Thoughts?